### ISSUE TITLE
How to swith language with slug (using Cviebrock sluggable)

### ISSUE THREAD

**Issue by megavolkan** [codezero-be/laravel-localized-routes#11]
Hi, thanks for great project.
I'm using `Route::macro...` in AppServiceProviders to switch locale for current page with links like 'EN','DE','NL'. And it works great for static routes with translated parameters.
My question is how can i switch language and at the same time translate localized uri segment. For slugs i'm using cviebrock/eloquent-sluggable.
I'm trying to translate this uri: 
`site.com/en/products/en-product-title-slug`
into this:
`site.com/de/produkte/de-produkte-titel-slug`

At the moment i'm getting this: 
`site.com/de/produkte/[ENGLISH SLUG]`,
`site.com/nl/produkt/[ENGLISH SLUG]`
etc.

I prefer a global solution for the entire project but open to all suggestions.

**Comment by ivanvermeyen** [comment]
Hey,

You're right, I didn't think of that.
It seems like you need to be able to pass in the parameters yourself by changing the macro a bit:

```php
public function boot()
{
    \Route::macro('currentUrl', function ($locale = null, $parameters = null) {
        return route(\Route::currentRouteName(), $parameters ?? \Route::current()->parameters(), true, $locale);
    });
}
```

Then you can do something like:

```php
route()->currentUrl('de', [$model->getSlug('de')]);
```

Or something in that genre?

**Comment by megavolkan** [comment]
Maybe a different approach would be returning not only one (current) url but all of them in an object/array available to the view. Like this:
```
$url[1] = www.site.com/en/products/en-product-name
$url[2] = www.site.com/de/produkte/de-produkte-name
$url[3] = www.site.com/fr/produits/fr-nom-de-produit
```
Then assign those to the links as language switcher.
I have no idea how to achieve that anyway :)

**Comment by ivanvermeyen** [comment]
That would also be a possibility, but it would still require some reference to whatever model you are using at that time... So same problem I'm afraid :/

**Comment by ivanvermeyen** [comment]
Could this be related to issue #13?
Will try to put this under test.

**Comment by ivanvermeyen** [comment]
I have created a Laravel 6 app to demonstrate the basic usage of this package.
I also wrote some tests for the URL switcher macro.
Take a look at the commit history to see what has been added.
Hope it helps.

https://github.com/ivanvermeyen/test-app-laravel-localized-routes 

Also, feel free to open any issues on the demo app if you want me to test/demo more stuff.
This helps me to improve the package too.
